apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -816,9 +816,9 @@ repack(Start, End, NextCursor, RightBound, RequiredPacking, StoreID) ->
 												{register_packing_ref, Ref, PaddedOffset}),
 										ar_util:cast_after(300000, Server,
 												{expire_repack_request, Ref}),
-										ar_events:send(chunk, {repack_request, Ref,
+										ar_packing_server:request_repack(Ref,
```

Comment:
Since only the `ar_packing_server` handles the `chunk, repack_request` and `chunk, unpack_request` events, we'll send the message straight to the `ar_packing_server` rather than broadcasting it to all `chunk` subscribers. This cuts down on the number of messages that can queue up in different processes' mailboxes (and occasionally lead to OOM exceptions)

Reply:
I would go even further and reply via direct messages too (reply PID may be passed to the request_* functions)

Reply:
Done! https://github.com/ArweaveTeam/arweave/pull/429/commits/6937c0c5540055bfe5ba92eccdfdba2d14e16101

I have one question called out here: https://github.com/ArweaveTeam/arweave/pull/429/files#r1230125676